lib/SIL/SILPrinter.cpp

```diff
@@ -3057,6 +3057,51 @@ void SILDefaultWitnessTable::dump() const {
   print(llvm::errs());
 }
 
+// TODO(TF-893): Use this helper to dedupe the same logic in
+// `SILFunction::print`.
+static void printSILFunctionNameAndType(
```

Comment:
Note: these printer changes were necessary to fix round-trip printing/parsing errors for generic signature requirements:
```
<stdin>:1864:88: error: use of undeclared type 'T'
sil_differentiability_witness hidden [parameters 0 1] [results 0]  [where T : _Differentiable]  @generic : $@convention(thin) <τ_0_0> (@in_guaranteed τ_0_0, Float) -> @out τ_0_0 {
```

Copying logic from `SILFunction::print` was the only way I found to fix the errors (by printing the generic signature requirements in the original function's generic environment).

Reply:
Actually, I found out how to print `τ_0_0 `: https://gist.github.com/dan-zheng/96661bc958f05374a2615f2b7aac5867.

Will keep printing `T` since it is the output of SILGen.